internal/resources/iampolicy/resource_iam_policy.go

```diff
@@ -0,0 +1,515 @@
+				return nil, errors.Wrapf(err, "unable to get Role Bindings for namespace")
+			}
+
+			policyList = resp.PolicyList
+		}
+	case unknownScope:
+		return nil, errors.Wrapf(err, "unable to get Role Binding; No valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scopesAllowed[:], `, `))
+	}
+
+	return policyList, err
+}
+
+// resourceIAMPolicyRead returns the intersection between binding list in terraform state and TMC server.
+func resourceIAMPolicyRead(_ context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config, _ := m.(authctx.TanzuContext)
+
+	var (
+		meta         *objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta
+		rbServerList []*iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding
+		calRBList    []*iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding
+	)
+
+	scopedFullnameData := constructScope(d)
+	rbStateList := constructRoleBindingList(d)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("unable to get Role Bindings; Scope full name is empty")
+	}
+
+	policyList, err := retrieveRoleBindingListFromServer(config, scopedFullnameData)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	// when iam policy resource is empty: no role bindings are existing, this is equivalent to not found condition.
+	if len(policyList) == 0 {
+		d.SetId("")
+		return
+	}
+
+	d.SetId(d.State().ID)
+
+	if err := d.Set(common.MetaKey, common.FlattenMeta(meta)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// iterate over the policy lists: state and server, and store the intersection of the two
+	for _, policy := range policyList {
+		if policy.Meta.UID == d.State().ID {
+			rbServerList = append(rbServerList, policy.RoleBindings...)
+		}
+	}
+	// nested iteration for preserving order of role binding lists.
+	for _, stateRB := range rbStateList {
+		for _, serverRB := range rbServerList {
+			if stateRB.Role == serverRB.Role {
+				calRBList = append(
+					calRBList,
+					&iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding{
+						Role:     serverRB.Role,
+						Subjects: getIntersectionOfSubs(stateRB.Subjects, serverRB.Subjects),
+					},
+				)
+			}
+		}
+	}
+
+	if err := d.Set(roleBindingsKey, flattenRoleBindingList(calRBList)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diags
+}
+
+func getIntersectionOfSubs(
+	state, server []*iammodel.VmwareTanzuCoreV1alpha1PolicySubject,
+) []*iammodel.VmwareTanzuCoreV1alpha1PolicySubject {
+	var newList []*iammodel.VmwareTanzuCoreV1alpha1PolicySubject
+	// nested iteration for preserving order of subjects.
+	for _, each := range state {
+		for _, sub := range server {
+			if sub.Name == each.Name && *sub.Kind == *each.Kind {
+				newList = append(newList, each)
+			}
+		}
+	}
+
+	return newList
+}
+
+func getMapToUpdate(rbl []*iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding, rblTypeVal rblType, subjectIntersect *map[string]iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpType) {
```

Comment:
here you are not returning a map,  so can you rename the function into constructRBOpForUpdate